bot/models/class_models.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from datetime import datetime
+
+from bot.utils.helpers import strtodt
+
+
+@dataclass
+class ClassSemester:
+    semester_id: str
+    semester_name: str
+    semester_start: str
+    semester_end: str
+
+    def start_date(self) -> datetime:
+        return strtodt(self.semester_start)
+
+    def end_date(self) -> datetime:
+        return strtodt(self.semester_end)
+
+
+@dataclass
+class ClassChannelScaffold:
+    class_prefix: str
+    class_number: int
+    class_professor: str
+    class_name: str
+
+    def class_code(self) -> str:
+        return f'{self.class_prefix}-{self.class_number}'
+
+    def full_title(self) -> str:
+        return f'{self.class_code()}: {self.class_name}'
+
+    def channel_name(self) -> str:
+        return f'{self.class_code().lower()}-{self.class_professor.lower()}'
```

Comment:
Why not make these properties?

Reply:
Because they are read-only and are just for formatting.

Reply:
Wroooong

```py
@property
def bruh():
   return "This shit unsettable"
```